translator/source_code/translator_shared_library.py

```diff
@@ -0,0 +1,95 @@
+import xmltodict
+import json
+from datetime import datetime
+import uuid
+import random
+import string
+import sys, getopt
+from jsonschema import validate
+from jsonschema import ValidationError
+import logging
+from collections import OrderedDict
+import re
+import os
+
+
+
+def validate_info(info):
+
+    if ((not info) or (type(info) != dict and type(info) != OrderedDict)):
+        logging.warning('invalid type')
+        return False
+
+    feed_info_id = str(info.get('feed_info_id', ''))
+    check_feed_info_id = re.match('[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}', feed_info_id)
+    
+
+    metadata=info.get('metadata', {})
+    wz_location_method = metadata.get('wz_location_method')
+    lrs_type = metadata.get('lrs_type')
+    contact_name = metadata.get('contact_name')
+    contact_email = metadata.get('contact_email') 
+    issuing_organization=metadata.get('issuing_organization')
+    required_fields = [ check_feed_info_id, metadata, wz_location_method, lrs_type, contact_name, contact_email, issuing_organization]
+    for field in required_fields:
+        if not field:
+            logging.warning( 'Not all required fields are present') 
+            return False
+            
+    return True
+
+def parse_xml(inputfile):
+    with open(inputfile, encoding='utf-8-sig') as ficone:
+        xml_string = ficone.read()
+        icone_obj = xmltodict.parse(xml_string)
```

Comment:
Since this is for more than icone now, let's rename those variables accordingly.

Reply:
should be fixed now